acceptance/tests/ensure_hiera_config_version.rb

```diff
@@ -0,0 +1,38 @@
+test_name 'C100304: Ensure hiera.yaml file contains version 5 format on new install' do
+
+  require 'puppet/acceptance/temp_file_utils'
+  extend Puppet::Acceptance::TempFileUtils
+
+  def environment_hiera_conf(agent)
+    if agent['platform'] =~ /windows/
+      'C:/ProgramData/PuppetLabs/code/environments/production/hiera.yaml'
+    else
+      '/etc/puppetlabs/code/environments/production/hiera.yaml'
+    end
+  end
+
+  def puppet_hiera_conf(agent)
+    if agent['platform'] =~ /windows/
+      'C:/ProgramData/PuppetLabs/puppet/etc/hiera.yaml'
+    else
+      '/etc/puppetlabs/puppet/hiera.yaml'
+    end
+  end
+
+  agents.each do |agent|
+    [puppet_hiera_conf(agent), environment_hiera_conf(agent)].each do |file|
+      step "puppet agent #{file} exists" do
+        if !file_exists?(agent, file)
+          fail_test("Failed to find file #{file} on agent #{agent}")
+        end
+      end
+
+      step "hiera.yaml contains 'version 5'" do
+        on(agent, "cat #{file}") do |cat_results|
```

Comment:
Is the `file_exists?` call needed if you are going to try and cat the file? You could add an assert in this step that STDERR should be empty and remove the extra ssh call to the SUT.